tests/Provider/ArrayGuardsProviderTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DotTest\Rbac\Guard\Provider;
+
+use Dot\Rbac\Guard\Guard\Factory;
+use Dot\Rbac\Guard\Guard\GuardInterface;
+use Dot\Rbac\Guard\Provider\ArrayGuardsProvider;
+use PHPUnit\Framework\MockObject\Exception;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ServerRequestInterface;
+
+class ArrayGuardsProviderTest extends TestCase
+{
+    protected ArrayGuardsProvider $subject;
+
+    protected Factory $mockFactory;
+
+    /**
+     * @throws Exception
+     */
+    public function setUp(): void
+    {
+        $this->mockFactory = $this->createMock(Factory::class);
+
+        $this->subject = new ArrayGuardsProvider(['guard_factory' => $this->mockFactory]);
+    }
+
+    public function testGetGuardsEmptyGuards()
```

Comment:
Please fix the following:
* add missing return type declaration
* fix RedundantCondition check